tests/func/test_checkout.py

```diff
@@ -603,3 +603,64 @@ def test_checkout_should_avoid_copy_for_existing_cache(
 
     assert "Removing before checkout" not in caplog.text
     assert get_inode(repo_dir.FOO) == foo_inode
+
+
+class TestCheckoutShouldCopyOnExistingLink(object):
+    def _test(self, repo_dir, dvc_repo, caplog, link_type):
+        ret = main(["config", "cache.type", link_type])
```

Comment:
You could also simply do
```
dvc_repo.cache.local.cache_types = [link_type]
```
instead of fiddling with reinits and CLI

Reply:
done